Lib/_pydatetime.py

```diff
@@ -500,14 +501,20 @@ def _parse_isoformat_time(tstr):
         else:
             tzsign = -1 if tstr[tz_pos - 1] == '-' else 1
 
-            td = timedelta(hours=tz_comps[0], minutes=tz_comps[1],
-                           seconds=tz_comps[2], microseconds=tz_comps[3])
-
-            tzi = timezone(tzsign * td)
+            try:
+                _check_time_fields(hour=tz_comps[0], minute=tz_comps[1],
```

Comment:
```suggestion
                # This function is intended to validate datetimes, but because
                # we restrict time zones to ±24h, it serves here as well.
                _check_time_fields(hour=tz_comps[0], minute=tz_comps[1],
```

I am mildly uncomfortable with this because it's kind of a canonical overuse of DRY (yes the code is the same *now*, but the function is doing something conceptually different, so you can't necessarily expect them to evolve in the same way). On balance I think we can get away with doing it anyway, but it would be good to leave a comment for future maintainers.